### PR TITLE
Added sanitizer-specific suffixes

### PR DESCRIPTION
**Pull request by Rubonnek**
I'd like to have separated Godot binary filenames that display which sanitizers were enabled.

Not sure if we want this, but I find it useful.

### CODE REVIEW

File: platform/server/detect.py [hunk 1]
```diff
@@ -107,23 +107,27 @@ def configure(env):
         env.Append(LINKFLAGS=["-ftest-coverage", "-fprofile-arcs"])
 
     if env["use_ubsan"] or env["use_asan"] or env["use_lsan"] or env["use_tsan"]:
-        env.extra_suffix += "s"
+        env.extra_suffix += "."
```
Comment (Xrayez) [inline platform/server/detect.py:110]: I think this `"."` could help a lot visually, currently compiling with `extra_suffix` and sanitizer support: #37301.

**Comment by aaronfranke** [comment]
@Rubonnek Is this still desired? If so, it needs to be [rebased](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase) and tested on the latest master branch, and the above feedback needs to be addressed. Otherwise, abandoned PRs will be closed.

**Comment by Rubonnek** [comment]
@aaronfranke Yes, I'd still like to have this change. I've update the PR. There's also nothing to address in the feedback above as it is a compliment since this PR helps a bit with #37301 but does not solve that issue.

**Comment by akien-mga** [comment]
A significant issue with this is that it will increase the disk space usage when doing multiple builds, since it will change the name of all intermediate compiled objects. So this would break incremental builds when changing from one sanitizer to another. It also makes it harder to guess what the binary name will be unless you know exactly in what order the options get parsed to append a suffix. WDYT @qarmin?

**Comment by qarmin** [comment]
I don't think that this will cause increased disk usage, because users(or just me) use most of the time only 1 set of sanitizers options (undefined + address + leak). Thread(very very slow) and Memory(for now barely usable in Godot) sanitizer cannot be mixed with previous options and are used very rarely so only a few people will need to use more space, but this will help with recompilation, as in the meantime the files will not be overwritten when using different set of options.

I'm not sure about incremental builds, because from my experience it doesn't work between two builds with different sanitizers options(`scons use_tsan=yes` overwrite all generated files from `scons use_asan=yes`). As for the suffixes, they will allow to see without running Godot what flags were used to create it. This is not problem for me, but maybe other may have a little problems with it.

**Comment by dmoody256** [comment]
> A significant issue with this is that it will increase the disk space usage when doing multiple builds, since it will change the name of all intermediate compiled objects. So this would break incremental builds when changing from one sanitizer to another. It also makes it harder to guess what the binary name will be unless you know exactly in what order the options get parsed to append a suffix. WDYT @qarmin?

Would it need to be intermediate files? Seems the original request is just related to the final binary.

- If the concern is disk space, don't change the intermediate suffix, and intermediates will be overwritten. Do change the `PROGSUFFIX` so the binary is built separately.
- If the concern is incremental build speed, use different suffixes for intermediates and the final binary.

Possibly those two ideas could be made into an option, but that seems a bit contrived.

**Comment by Rubonnek** [comment]
Closing pull request as I'm no longer interested in maintaining it.

I wrote some shell functions on my end as my workaround to renaming the binaries.

**Pull request closed without merging**